packages/sdk/src/evm/common/metadata-resolver.ts

```diff
@@ -60,11 +60,13 @@ export async function fetchContractMetadataFromAddress(
     );
     try {
       // try from multichain registry
-      const importedUri =
-        await polygonSDK.multiChainRegistry.getContractMetadataURI(
-          chainId,
-          address,
-        );
+      const contract = new Contract(
```

Comment:
we're constructing this every time here now but it is immutable, I think we should construct it the first time we need it and then keep it around instead

`let contract;` above the FN

```js

if(!contract){
  contract = /*{... the implementation here}*/
}
```

Reply:
done